Dockerfile

```diff
@@ -52,12 +52,22 @@ ENTRYPOINT [ "solvers" ]
 
 # Extract Binary
 FROM intermediate
+RUN apt-get update && \
```

Comment:
How much of a size difference does this make?
AFAICS this should add the entire repo, dependencies and build artifacts to the image.
Also what't the impact on build times for the image?

Reply:
The build time is the same: https://github.com/cowprotocol/services/actions/runs/9778536743 vs https://github.com/cowprotocol/services/actions/runs/9773882766
The image size is 1.93GB vs 1.22GB.

Reply:
Okay not too bad. Does the `heaptrack` also have a `make clean` command to remove the intermediate build artifacts?

Reply:
Added a clean-up logic.